### ISSUE TITLE
Render due dates as diagnostics

### ISSUE THREAD

**Issue by skbolton** [huantrinh1802/m_taskwarrior_d.nvim#92]
I noticed in the code that the virtual dates are rendered as a [custom highlight](https://github.com/huantrinh1802/m_taskwarrior_d.nvim/blob/d0cf7a7e882e308ab203d9973ce3147a328258f7/lua/m_taskwarrior_d/init.lua#L690). I was first going to suggest that you expose this highlight for themes to customize but as I was thinking about what I would set it to I figured my diagnostic highlights would be what I would want to use. Then it hit me that as diagnostics you wouldn't need to do any custom highlighting.

**Comment by huantrinh1802** [comment]
Hi @skbolton, are you preferring to this https://neovim.io/doc/user/diagnostic.html#hl-DiagnosticError and alike?

**Comment by skbolton** [comment]
Yes, but more specifically the whole diagnostic api in neovim. See `:help diagnostic.txt`

If you are unfamiliar with them its the api that neovim added so that linters, lsps, test runners, etc can diagnostic information to a given line. An example of another repo that uses the diagnostic api to do line decorations would be [todo-comments](https://github.com/folke/todo-comments.nvim)

**Comment by skbolton** [comment]
This would mean a lot less work for you. You only have to call the diagnostic api. Themes already define good highlights for these, icons in the gutter can be configured, and you can do things like open diagnostics in telescope or the quickfix list

**Comment by chuan2984** [comment]
I use mini files with preview, and when the render due date option is turned on, each rendering takes about 3 seconds, im assuming its because it suffers from the same problem as the `sync`
